crates/cargo-remove/src/cargo/ops/cargo_remove/manifest.rs

```diff
@@ -53,6 +53,14 @@ impl DepTable {
         self.kind
     }
 
+    pub fn kind_table_name(&self) -> &str {
+        match self.kind {
+            DepKind::Normal => "dependencies",
+            DepKind::Development => "dev-dependencies",
+            DepKind::Build => "build-dependencies",
+        }
+    }
```

Comment:
This doesn't really belong in this level of the abstraction.  It'd be great if `DepKind` could own this directly.

I was curious what cargo-add does and found it does the same index stuff.  I guess I turned a blind eye to that when cleaning up add.

Some possibilities
- We move forward with this PR as-is
- We punt on this and clean up both this and add at the same time, putting this function on `DepKind`
- We change the output to show the full table name.  

What are your thoughts?

Reply:
The full table name solution (e.g. `Removing docopt from target.x86_64-unknown-linux-gnu.build-dependencies`) isn't the worst, especially since it makes it easier to imagine the change made to Cargo.toml (I often will check it to make sure it looks right out of curiosity). Perhaps a little wordy though. The asymmetry between `cargo add` and `cargo remove` is more painful, so I would prefer that if we make that change, it be made later to both subcommands instead.

I suppose we could just add a comment to the cargo PR saying we'll fix it in a future release? Curious to know what you think.

Reply:
Actually, now that I think more about it, displaying the full table name will have its own API challenges, because then we will have to decide whether or not to escape the target. Eek

Reply:
> I suppose we could just add a comment to the cargo PR saying we'll fix it in a future release? 

Probably small enough not being worth bringing up in the PR description.  Sometimes I'll open comments on my own PRs and that could possibly be done but overall this feels like a low priority to me.

> displaying the full table name will have its own API challenges

A `Display` has a lower barrier for perfection, even more if we put it in a `fn display` (think `PathBuf`)